code/hdb/hdbstandard.q

```diff
@@ -1,7 +1,28 @@
 // reload function
 reload:{
+	if[.z.w in key .hdb.reloadcalls;
+		.hdb.reloadcalls[.z.w]:1b;
+		$[not all .hdb.reloadcalls; 
+			{.lg.o[`reload;"reload call received from handle ", string[.z.w], "; reload calls pending from handles ", ", "sv string where not .hdb.reloadcalls]; :(::)}[];
+			.lg.o[`reload;"reload call received from handle ", string[.z.w], "; no more reload calls pending"];
+		]
+	]
 	.lg.o[`reload;"reloading HDB"];
+	@[`.hdb.reloadcalls;key .hdb.reloadcalls;:;0b];
 	system"l ."}
 
 // Get the relevant HDB attributes
 .proc.getattributes:{`date`tables!(@[value;`date;`date$()];tables[])}
+
+\d .hdb
+
+// dictionary of handles to reload
+reloadcalls:()!();
+
+// function to add handle to reloadcalls dictionary
+po:{[h] if[.proc.proctype in .hdb.connectedProcs;reloadcalls[h]:0b]};
```

Comment:
`.proc.proctype` is the _current_ process's proctype i.e. here it will always be `hdb` - I don't think this is right